exonum/src/node/mod.rs

```diff
@@ -268,6 +269,51 @@ pub struct NodeConfig {
     pub thread_pool_size: Option<u8>,
 }
 
+impl NodeConfig<PathBuf> {
+    /// Converts `NodeConfig<PathBuf>` to `NodeConfig<SecretKey>` reading the key files.
+    pub fn read_secret_keys(
+        self,
+        config_file_path: impl AsRef<Path>,
+        consensus_passphrase: &[u8],
+        service_passphrase: &[u8],
+    ) -> NodeConfig {
+        let config_folder = config_file_path.as_ref().parent().unwrap();
+        let consensus_key_path = if self.consensus_secret_key.is_absolute() {
+            self.consensus_secret_key
+        } else {
+            config_folder.join(&self.consensus_secret_key)
+        };
+        let service_key_path = if self.service_secret_key.is_absolute() {
+            self.service_secret_key
+        } else {
+            config_folder.join(&self.service_secret_key)
+        };
+
+        let consensus_secret_key = read_keys_from_file(&consensus_key_path, consensus_passphrase)
+            .expect("Could not read consensus_secret_key from file")
+            .1;
+        let service_secret_key = read_keys_from_file(&service_key_path, service_passphrase)
+            .expect("Could not read service_secret_key from file")
+            .1;
+        NodeConfig {
+            consensus_secret_key,
+            service_secret_key,
+            genesis: self.genesis,
+            listen_address: self.listen_address,
+            external_address: self.external_address,
+            network: self.network,
+            consensus_public_key: self.consensus_public_key,
```

Comment:
What happens if the keypair in the key file doesn't correspond to this public key?

Reply:
Nothing right now. Should we throw an error in this case?

I really discovered that we store public key in the key files when I published the PR. I think that another option is to remove public keys from our final config and read them from key files, so we can get rid of redundancy and duplication of data. What do you think about it?